Change log output for failed namespace lookup

cpretzer opened the pull request:
## What
This PR changes the log output when retrieving the namespace from the `kubeconfig` during install and upgrade operations

## Why
As reported in #4818, if an error occurs while retrieving the namespace set for the current context, the error log message that is written doesn't print the underlying error nor does the message indicate that the `default` namespace will be used instead.

## How
Change the log message to `warn` level, print the underlying error from `client-go`, and report that the `default` namespace will be used

Signed-off-by: Charles Pretzer <charles@buoyant.io>

Matei207 commented:
This reads much better and is more informative than what I had originally put in there. Puts things into perspective for me, thank you!

cpretzer commented:
> I think we should make this message multiline instead of separating it into two `Warnf`s.

@kleimkuhler had that same thought initially, and changed it because I convinced myself it was more readable. Just updated with your suggestion 😄

cpretzer commented:
@kleimkuhler TIL about backticks preserving line breaks. Thanks! Update committed
